### ISSUE TITLE
[BUG] Terraform provider - StorageClassName is not respected when providing both image and StorageClass

### ISSUE THREAD

**Issue by vulkoingim** [harvester/harvester#2842]
**Describe the bug**

As the title says, when providing a StorageClassName, alongside an Image - the storage class for the PVC gets overwritten by the one derived from the image.

I already opened a PR on the [terraform-provider-harvester](https://github.com/harvester/terraform-provider-harvester) repo, but since no one got assigned automatically, I'm opening an issue here as well with the hope of resolving this soon :)

https://github.com/harvester/terraform-provider-harvester/pull/53

**To Reproduce**

Apply an example resource, either a volume or a VM with an inline definition for a disk with both the `image` and `storage_class_name` and the PVC will be created with the `storage_class` derived from the image.

```tf
resource "harvester_volume" "ex" {
  name = "ex"
  size = "200Gi"

  image              = "longhorn"
  access_mode        = "ReadWriteOnce"
  volume_mode        = "Block"
  storage_class_name = "your-storage-class"
}
```

**Expected behavior**

PVC gets created with the supplied `storage_class_name`. This is the standard behaviour, if a VM is created with a YAML manifest.

**Comment by futuretea** [comment]
working on it: https://github.com/harvester/harvester/issues/2793

**Comment by futuretea** [comment]
Fixed in Harvester v1.1.0 and terraform-provider-harvester v0.6.0

**Comment by futuretea** [comment]
close in favor of terraform-provider-harvester v0.6.0
